### ISSUE TITLE
[Static Web Assets] DefineStaticWebAssetEndpoints throws ArgumentException due to duplicate key

### ISSUE THREAD

**Issue by martincostello** [dotnet/sdk#43033]
### Describe the bug

Building a Blazor WASM project fails due to an `ArgumentException` being thrown by the `DefineStaticWebAssetEndpoints` task.

Looks similar to some changes made in #41681.

### To Reproduce

- Clone https://github.com/martincostello/benchmarks-dashboard/pull/2/commits/01f6219f9eb7e7b83c0f25d46bac45355ba4ce37
- Run `build.ps1` from the root of the repository.

### Exceptions (if any)

```console
/usr/share/dotnet/sdk/9.0.100-preview.7.24407.12/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(326,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018: The "DefineStaticWebAssetEndpoints" task failed unexpectedly. [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018: System.ArgumentException: An item with the same key has already been added. Key: /home/runner/work/benchmarks-dashboard/benchmarks-dashboard/artifacts/bin/Dashboard/release/wwwroot/_framework/Humanizer.resources.wasm [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior) [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value) [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer) [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssetEndpoints.Execute() [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute() [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
/home/runner/.nuget/packages/microsoft.net.sdk.webassembly.pack/9.0.0-preview.7.24405.7/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets(347,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/runner/work/benchmarks-dashboard/benchmarks-dashboard/src/Dashboard/Dashboard.csproj]
Exception: /home/runner/work/benchmarks-dashboard/benchmarks-dashboard/build.ps1:98
```

### Further technical details

.NET SDK `9.0.100-preview.7.24407.12`

**Comment by martincostello** [comment]
I've tried this workaround, but it doesn't work: https://github.com/dotnet/aspnetcore/issues/57512#issuecomment-2310498743

**Comment by pavelsavara** [comment]
cc @javiercn @maraf

**Comment by jeromelaban** [comment]
As mentioed in https://github.com/dotnet/aspnetcore/issues/57512#issuecomment-2310498743, this is fixed at least in 9.0.100-rc.2.24428.9.

**Comment by martincostello** [comment]
Just tested and it does appear to be fixed with version `9.0.100-rc.2.24429.6` of the SDK.

**Comment by jeromelaban** [comment]
@javiercn would you know if the fixes from https://github.com/dotnet/aspnetcore/issues/57512 will be included in RC1? I could not find any related PR that indicates what change was made to fix this issue.
